extensions/amp-bind/amp-bind.md

```diff
@@ -66,425 +72,972 @@ The `amp-bind` component allows you to add custom stateful interactivity to your
+  - Custom functions, classes and loops are disallowed.
+- Undefined variables and array-index-out-of-bounds return `null` instead of `undefined` or throwing errors.
+- A single expression is currently capped at 50 operands for performance. Please [contact us](https://github.com/ampproject/amphtml/issues/new) if this is insufficient for your use case.
+
+The following are all valid expressions:
+
+[example preview="inline" playground="true" imports="amp-bind"]
+
+```html
+<p [text]="myExpressionsState.foo"></p>
+<!-- 1 + '1'; // 11 -->
+<button on="tap:AMP.setState({myExpressionsState: {foo: 1 + '1'}})">
+  foo: 1 + "1"
+</button>
+<!-- 1 + +'1'; // 2 -->
+<button on="tap:AMP.setState({myExpressionsState: {foo: 1 + + '1'}})">
+  foo: 1 + + "1"
+</button>
+<!-- !0; // true -->
+<button on="tap:AMP.setState({myExpressionsState: {foo: !0}})">foo: !0</button>
+<!-- null || 'default'; // 'default' -->
+<button on="tap:AMP.setState({myExpressionsState: {foo: null || 'default'}})">
+  null || "default"
+</button>
+<!-- [1, 2, 3].map(x => x + 1); // 2,3,4 -->
+<button
+  on="tap:AMP.setState({myExpressionsState: {foo: [1, 2, 3].map(x => x + 1)}})"
+>
+  [1, 2, 3].map(x => x + 1)
+</button>
+```
+
+[/example]
+
+Find the full expression grammar and implementation in [bind-expr-impl.jison](./0.1/bind-expr-impl.jison) and [bind-expression.js](./0.1/bind-expression.js).
+
+### Allowed-listed functions
+
+#### [`Array`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array#Methods)
+
+Single-parameter arrow functions can't have parentheses, e.g. use `x => x + 1` instead of `(x) => x + 1`. `sort()` and `splice()` return modified copies instead of operating in-place.
+
+- [concat](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/concat)
+- [filter](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter)
+- [includes](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes)
+- [indexOf](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/indexOf)
+- [join](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/join)
+- [lastIndexOf](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/lastIndexOf)
+- [map](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map)
+- [reduce](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce)
+- [slice](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/slice)
+- [some](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some)
+- [sort](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort) (not-in-place)
```

Comment:
This says `not-in-place` but there is an example for it below, should this omit `not-in-place`?